File: java/mixedfractions.java
import java.util.Scanner;

public class mixedfractions {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int numerator, denominator;
		while ((numerator = scan.nextInt()) != 0 | (denominator = scan.nextInt()) != 0)
			System.out.printf("%d %d / %d", numerator / denominator, numerator % denominator, denominator);
		scan.close();
	}
}

File: ts/src/mixedfractions.ts
import readline = require('readline')

function extract (numerator: number, denominator: number): [number, number] {
  return [Math.floor(numerator / denominator), numerator % denominator]
}

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (line === '0 0') {
      process.exit(0)
    }
    const [numerator, denominator] = line.split(' ').map(Number)
    console.log(
      `${extract(numerator ?? 0, denominator ?? 0).join(' ')} / ${denominator}`
    )
  })
  .on('close', () => {
    process.exit(0)
  })
